layers/igmp.go

```diff
@@ -16,18 +16,210 @@ import (
 
 type IGMPType uint8
 
+const (
+	MembershipQuery    IGMPType = 0x11 // General or group specific query
+	MembershipReportV1 IGMPType = 0x12 // Version 1 Membership Report
+	MembershipReportV2 IGMPType = 0x16 // Version 2 Membership Report
+	LeaveGroup         IGMPType = 0x17 // Leave Group
+	MembershipReportV3 IGMPType = 0x22 // Version 3 Membership Report
+)
+
+// String conversions for IGMP message types
+func (i IGMPType) String() string {
+	switch i {
+	case MembershipQuery:
+		return "IGMP Membership Query"
+	case MembershipReportV1:
+		return "IGMPv1 Membership Report"
+	case MembershipReportV2:
+		return "IGMPv2 Membership Report"
+	case MembershipReportV3:
+		return "IGMPv3 Membership Report"
+	case LeaveGroup:
+		return "Leave Group"
+	default:
+		return ""
+	}
+}
+
+type IGMPQuery interface {
+	decodeResponse(data []byte) error
+}
+
 // IGMP is the packet structure for IGMP messages.
 type IGMP struct {
 	BaseLayer
-	Type            IGMPType
-	MaxResponseTime time.Duration
-	Checksum        uint16
-	GroupAddress    net.IP
-	// The following are used only by IGMPv3
-	SupressRouterProcessing bool
-	RobustnessValue         uint8
-	IntervalTime            time.Duration
-	SourceAddresses         []net.IP
+	Type    IGMPType  // IGMP message type
+	Message IGMPQuery // Container for message payload
+	Version uint8     // IGMP protocol version
+}
+
+type GroupRecordType uint8
+
+const (
+	IsIn  GroupRecordType = 0x01 // Type MODE_IS_INCLUDE, source addresses x
```

Comment:
The 'layers' package contains a namespace used by a whole ton of different protocols.  As such, please prefix any protocol-related objects with the name of the protocol:

IGMPGroupRecordType
IGMPIsIn

etc?